pkg/management/barman/commandbuilder.go

```diff
@@ -74,6 +89,12 @@ func appendCloudProviderOptions(
 				options,
 				"--cloud-provider",
 				"azure-blob-storage")
+			if azureInheritFromAzureAD {
```

Comment:
@valeriodelsarto shouldn't we be checking for barman version `2.18` and above as it was only introduced only @ 2.18? 
https://github.com/EnterpriseDB/barman/releases

Reply:
You're right @dhilipkumars, I haven't check for that but here we need to be sure that we are using an image which have Barman version 2.18 or above. We need to add this check in this feature branch.

Reply:
I've added the check, waiting for a review. Thanks!

Reply:
I've also added `//nolint: nestif` because the linter was complaining about the complexity of the nested if statements here, feel free to refactor it removing this